### PR TITLE
Scalable component library

### LINKED ISSUE #503: Implement scalable component library
__What should we do?__

- [ ] As a form designer, I expect the most commonly used components to appear at the top, so I don't need to scroll to use them.
- [ ] I want to search for components by name.
- [ ] I want to browse all available components.
- [ ] I expect components to be categorized by use case (such as "Basic Input", "Selection", "Presentation", "Action"...).
- [ ] (Stretch) I want to browse and add components to the end of the form using my keyboard.
- [ ] (Stretch) I want to have different UI break points (screen size, playground mode)
- [ ] [Integrate into `form-playground`](https://github.com/camunda/form-playground/issues/49)

image

__Why should we do it?__

We want to help form designers build forms in less time by providing component library that allows to easily discover, browse, and use form components, as well as to understand what these components are used for.

* Check out the [product hub epic](https://github.com/camunda/product-hub/issues/728) to gather more rationale.
* [Competitor analysis](https://miro.com/app/board/uXjVP8rk-qI=/)
* [Mockups](https://www.figma.com/file/CiTSN1vNETjIAy6W9jUIFp/Form-Editor?node-id=723%3A43296&t=T4tVSiG5XZRNoZJa-0)
* [Kick off Miro board](https://miro.com/app/board/uXjVPx8pfPU=/)

Related to https://github.com/camunda/product-hub/issues/728

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #503 
Related to https://github.com/camunda/form-playground/pull/50

Try out the demo: Demo: https://49-updated-palette--camunda-form-playground.netlify.app/

image

image

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
To be considered: updating the dragging state

image

**Review by vsgoulart — COMMENTED** [review]
Looks good! Just some small comments

**Review by vsgoulart — APPROVED** [review]

File: packages/form-js-editor/assets/form-js-editor.css [hunk 1]
```diff
@@ -15,13 +15,27 @@
 }
 
 .fjs-palette-container {
-  --color-palette-container-background: var(--color-grey-225-10-97);
+  --color-palette-text: var(--color-grey-225-10-15);
```
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply packages/form-js-editor/assets/form-js-editor.css:18]: :+1: Let's have an issue to start a conversation. I think we can learn a lot from Carbon here in terms of customization!

[the remark this answers is not in the document]
Comment (vsgoulart) [reply packages/form-js-editor/assets/form-js-editor.css:18]: Done #515

**Comment by pinussilvestrus** [comment]
@RomanKostka I updated the drag preview as you proposed. I like 👍 

Kapture 2023-01-27 at 15 06 11

**Comment by pinussilvestrus** [comment]
Let me know when there is something else. I would merge it otherwise 👍

**Comment by christian-konrad** [comment]
A quick thing: we should add `user-select: none`, while dragging, I accidently selected all the labels:
Bildschirmfoto 2023-01-27 um 18 44 15

**Review by christian-konrad — CHANGES_REQUESTED** [review]
Unfortunately, it is now cropping the palette already under 1500px:

Bildschirmfoto 2023-01-27 um 19 05 06

**Comment by Skaiir** [comment]
> @RomanKostka I updated the drag preview as you proposed. I like 👍

Now onto dynamically calculating the angle of the transform based on the acceleration of the cursor :trollface:

**Comment by pinussilvestrus** [comment]
> Unfortunately, it is now cropping the palette already under 1500px:

I will adjust this via https://github.com/camunda/form-playground/pull/50

**Comment by pinussilvestrus** [comment]
@RomanKostka @christian-konrad, the PR should be ready now. Let me know if there is anything left or whether we can merge it 👍

**Review by christian-konrad — APPROVED** [review]
lgtm

**Comment by pinussilvestrus** [comment]
Merged it now, as something's going on with CodeCov (unrelated to this PR)

🎉

**Pull request merged**